### PR TITLE
*: add tiflash_hashagg_preaggregation_mode sys var

### LINKED ISSUE #54867: support tiflash_hashagg_preaggregation_mode
## Enhancement
this is part of tiflash `auto pass through hashagg` feature. ref https://github.com/pingcap/tiflash/issues/9196

### PR DESCRIPTION
**Pull request by guo-shaoge**
### What problem does this PR solve?

Issue Number: close #54867

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -1501,6 +1503,7 @@ const (
 	DefTiDBDMLType                                    = "STANDARD"
 	DefGroupConcatMaxLen                              = uint64(1024)
 	DefDefaultWeekFormat                              = "0"
+	DefTiFlashPreAggMode                              = "auto"
```
Comment (guo-shaoge) [inline pkg/sessionctx/variable/tidb_vars.go:1506]: todo verify "auto" or "force_preagg"

File: pkg/planner/core/task.go [hunk 2]
```diff
@@ -2235,6 +2238,9 @@ func (p *PhysicalHashAgg) attach2TaskForMpp(tasks ...base.Task) base.Task {
 			newMpp := mpp.enforceExchanger(exProp)
 			attachPlan2Task(middle, newMpp)
 			mpp = newMpp
+			if partialHashAgg, ok := partial.(*PhysicalHashAgg); ok && len(partitionCols) != 0 {
+				partialHashAgg.tiflashPreAggMode = p.SCtx().GetSessionVars().TiFlashPreAggMode
+			}
```
Comment (windtalker) [inline pkg/planner/core/task.go:2243]: For some agg, it may generate 3 stage aggregation, will the first 2 stage agg be treated as partial agg?

Comment (guo-shaoge) [reply pkg/planner/core/task.go:2243]: I checked the 3-stage agg and found most 2nd agg has no group by column. So I think the 2nd agg no need to use auto pass through

**Review by winoros — APPROVED** [review]
planner part LGTM

**Review by windtalker — APPROVED** [review]
lgtm

**Pull request merged**